eventrouter/internal/list.c

```diff
@@ -0,0 +1,58 @@
+#include "list.h"
```

Comment:
Need `eventrouter_` prefix for file name?

Reply:
The C preprocessor searches `#include`ed files next to the source file first. I'll admit that it's not as clear as project-qualified paths, but this is equivalent to `#include "eventrouter/internal/list.h"`. If it would help, I can transform all "local" include names to project-qualified ones instead.